### ISSUE TITLE
Parser Error Handler Throws Exception for incomplete Source

### ISSUE THREAD

**Issue by nadeeshaan** [ballerina-platform/ballerina-lang#25313]
**Description:**
Consider the following source snippet

```
import ballerina/io;
import ballerina/http;

public const int HELLO = 12;

public function main() {
    e
    io:println(testArr.length());
    
}

type HelloErr error<map<string>>;

```

the above source snippet throws the following exception

```
java.util.NoSuchElementException
	at java.util.ArrayDeque.removeFirst(ArrayDeque.java:285)
	at java.util.ArrayDeque.pop(ArrayDeque.java:522)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.endContext(AbstractParserErrorHandler.java:209)
	at io.ballerinalang.compiler.internal.parser.AbstractParser.endContext(AbstractParser.java:126)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseCompUnit(BallerinaParser.java:175)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parse(BallerinaParser.java:94)
	at io.ballerinalang.compiler.syntax.tree.SyntaxTree.from(SyntaxTree.java:55)
	at io.ballerinalang.compiler.syntax.tree.SyntaxTree.from(SyntaxTree.java:50)
	at org.ballerinalang.langserver.compiler.workspace.WorkspaceDocument.<init>(WorkspaceDocument.java:44)
	at org.ballerinalang.langserver.compiler.workspace.WorkspaceDocument.<init>(WorkspaceDocument.java:50)
	at org.ballerinalang.langserver.compiler.workspace.WorkspaceDocumentManagerImpl.openFile(WorkspaceDocumentManagerImpl.java:89)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.didOpen(BallerinaTextDocumentService.java:664)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Affected Versions:**
Swan Lake Preview3 at least

**Comment by lochana-chathura** [comment]
Issue is from parsing of functional binding pattern. Exception should be gone away with PR #25521.

**Comment by rdulmina** [comment]
This issue requires a change to `BLangErrorVariable` node i.e: error cause allows error binding pattern in the spec, but `BLangErrorVariable.cause` is a `BLangSimpleVariable`. @SupunS WDYT?

**Comment by lochana-chathura** [comment]
> This issue requires a change to `BLangErrorVariable` node i.e: error cause allows error binding pattern in the spec, but `BLangErrorVariable.cause` is a `BLangSimpleVariable`. So will not be fixed with #25521. @SupunS WDYT?

I think it is better to fix it in a separate PR.

**Comment by SupunS** [comment]
+1 to fix it separately
